apollo-router/src/plugins/telemetry/endpoint.rs

```diff
@@ -64,7 +64,9 @@ impl UriEndpoint {
 
             Uri::from_parts(parts)
                 .expect("uri cannot be invalid as it was constructed from existing parts")
-        })
+        } else {
+            default_endpoint.clone()
```

Comment:
It's a main change, rest is just refactoring after removing Option wrapper

Reply:
Also, we already have this code in other places so it make sense to move it into a common function.

https://github.com/apollographql/router/blob/cb40e7a70429c4a279ffdddc436a9e3376640822/apollo-router/src/plugins/telemetry/tracing/jaeger.rs#L133-L134

Also it's weird that stuff like port number (if omitted) is populated from default but if you omit the entire URI or provide `default` is is populated by OLTP library and from Router's default.